### ISSUE TITLE
怎么把webpack的配置生成好自己维护

### ISSUE THREAD

**Issue by jiangzm** [sorrycc/roadhog#681]
roadhog在不满足需求的情况下希望能完全控制webpack的配置， 那么有没有类似create-react-app中eject的命令呢？
看到挺多有roadhog配置问题困惑的人，希望roadhog在集成到系统是个可选项而不是单一项。

**Comment by HiMrHu** [comment]
对于这个问题,首先这一类工具就是不希望用户私自更改配置,而将所有的优化配置都交由工具作者来完成,roadhog又提供了配置文件,可以让用户在基础上进行功能增强,比如插件/生产开发环境下配置,都可以在`.webpackrc.js`中进行设置,至于你说的,希望完全掌控webpac配置,那么首先需要对webpack有一个比较深的理解,同时如果有一个比较深的理解,那么为什么不自己配置一个呢,看着自己心仪的工具怎么配置照着配置一个webpack也是可以的,这样就玩却解决了您的问题.没有一个工具是通用的.roadhog只是作者很努力的希望给不懂webpack的使用者一个开箱即用的工具.

**Comment by jiangzm** [comment]
@HiMrHu  我清楚也看到roadhog做了很多适配的工作，我自己也写了一份ant-pro的webpack的配置，但是和roadhog编译生成的文件大小有差异。
“不希望用户私自更改配置”  不是很清楚这个私自是怎么定义的

**Comment by arasdean-work** [comment]
I completely agree with @jiangzm. I understand that Roadhog is a tool that provides the .webpackrc file for configuration, but webpack is blackboxed inside of af-webpack. While someone very strong in webpack could configure a config file themselves, it's much easier to just use a built-in command, like 'eject' provided in create-react-app, and then tweak the finer details of the config.

**Comment by arasdean-work** [comment]
I completely agree with @jiangzm. I understand that Roadhog is a tool that provides the .webpackrc file for configuration, but webpack is blackboxed inside of af-webpack. While someone very strong in webpack could configure a config file themselves, it's much easier to just use a built-in command, like 'eject' provided in create-react-app, and then tweak the finer details of the config.
